impl/client/src/main/javascript/web/pentaho/visual/views/echarts/Gauge.js

```diff
@@ -0,0 +1,142 @@
+
+      return [
+        {
+          type: "gauge",
+          min: 0,
+          max: Math.ceil(range.max / div) * div,
+
+          pointer: {
+            width: 8,
+            length: "80%",
+            offsetCenter: [0, "8%"]
+          },
+          progress: {
+            show: true,
+            overlap: true,
+            roundCap: true
+          },
+          detail: {
+            width: "auto",
+            height: "auto",
+            legendHoverLink: true,
+            fontSize: 14,
+            color: "#fff",
+            backgroundColor: "inherit",
+            borderRadius: 3,
+            formatter: "{value}"
+          },
+          itemStyle: {
+            borderColor: "#fff",
+            borderWidth: 1
+          },
+          emphasis: {
+            label: {
+              fontSize: 20
+            }
+          }
+        }
+      ];
+    },
+
+    /**
+     * This method transforms the table into a plain structure,
+     * obtains the required values for categories and measures from te source dataTable object,
+     * transforms them into target name-value pairs in string and number formats,
+     * handles their layout within the available width
+     * and builds the tooltip.
+     *
+     */
+    _buildData: function() {
+      var dataTable = this.model.data;
+
+      if(dataTable.originalCrossTable) {
+        dataTable = dataTable.originalCrossTable.toPlainTable({skipRowsWithAllNullMeasures: true});
+      }
+
+      var rowCount = dataTable.getNumberOfRows();
+      var categoriesColIndexes = this.model.rows.fieldIndexes;
+      var measureColIndex = this.model.measures.fieldIndexes[0];
+      var font = util.getDefaultFont(null, 12);
+
+      var records = [];
+
+      var maxWidth = this.model.width;
+      var midIdx = Math.floor(rowCount / 2);
+
+      // Midpoint x-coordinate
+      var initX = 0;
+      // Constant y-coordinate in percentage
+      var yPercentage = 100;
+
+      // Dynamic separation between elements
+      var hSeparation = (maxWidth / (rowCount - 1)) * 0.8;
+
+      // Read data from dataTable and push it to records
+      for(var i = 0; i < rowCount; i++) {
+        var offsetX;
+
+        if(rowCount % 2 === 0) {
+          if(i < midIdx) {
+            offsetX = -((midIdx - 0.5 - i) * hSeparation);
+          } else {
+            offsetX = (i - midIdx + 0.5) * hSeparation;
+          }
+        }else {
+          if(i === midIdx) {
+            offsetX = 0;
+          } else if(i < midIdx) {
+            offsetX = -((midIdx - i) * hSeparation);
+          } else {
+            offsetX = (i - midIdx) * hSeparation;
+          }
+        }
+
+        records.push({
+          name: dataTable.getFormattedValue(i, categoriesColIndexes),
+          value: dataTable.getValue(i, measureColIndex),
+          tooltip: this._buildTooltip(this._buildRowTooltipHtml(dataTable, i), font),
+          title: {
+            offsetCenter: [initX + offsetX, yPercentage + "%"],
```

Comment:
The code for determining the X position of the title and detail seems more complex that it needs to be.
For an equivalent behavior, you could try something like this:

```js
// Use a 10% margin on either side.
var innerWidth = this.model.width * 0.8;

var rowCount = dataTable.getNumberOfRows();

// Split space equally by all rows.
// Points are flushed on the sides. Horizontal separation only applied between points,
// given there's already an outer margin (of 10% on either side).
// Horizontally, title and detail are placed centered on these points.
// For first and last points, the outer margins absorb half of the title / detail. 
// May not fit in extreme cases.
var hSeparation = innerWidth / (rowCount - 1);

// Avoid overlapping title text by having a safety margin on either side of text (2.5%).
var titleWidth = hSeparation * 0.95;
var offsetCenterYPercent = 100;

for(var i = 0; i < rowCount; i++) {
  // X coordinate of title/detail relative to the center of the chart.
  // For use in option `offsetCenter`.
  var offsetCenterX = -innerWidth / 2 + hSeparation * i;

  records.push({
    // ...
    
    title: {
      offsetCenter: [offsetCenterX, offsetCenterYPercent + "%"],
      overflow: "truncate",
      width: titleWidth
    },
    detail: {
      offsetCenter: [offsetCenterX, (offsetCenterYPercent + 15) + "%"],
      width: titleWidth
    }
  });
}
```

However, note how this layout logic may cause the first and last labels to be clipped out of view in extreme cases. To avoid this, the space would need to be split using a different layout that take the width allocated for each row into account, instead of flushing points to the sides.

```js

/**
 * SPLIT BAND FLUSHED LAYOUT
 * 
 * Split space equally by N elements, into equally spaced bands, of width `B`.
 * Bands are separated by a margin `m`, and are flushed to the sides 
 * (given there's already an outer margin of 10% on either side).
 * By definition, a step is the difference between the starting points of two 
 * consecutive bands: `S = B + m`.
 * Horizontally, title and detail are actually placed centered in bands.
 * The proportion of a step which is occupied by a band is `b`: 
 * - `B = b * S`
 * - `m = (1 - b) * S`
 * 
 * Given these definitions, the flushed band layout looks like:
 * ====
 * | B m B m B | (m)
 *   S-> S-> S->
 * ====
 * 
 * Note that the last step overflows the container by `m`, on the right.
 * From which we can write the equation:
 * - `width = N * S - m`
 * 
 * Substituting the formula for `m`, above, results in:
 * - `S = width / (N - (1 - b))`
 * 
 * The starting position of a band `i` is:
 * - `start[i] = S * i`
 */

// Use a 10% margin on either side.
var innerWidth = this.model.width * 0.8;

var rowCount = dataTable.getNumberOfRows();

// Avoid overlapping title text by having a safety margin on either side of text (2.5%).
var bandPercent = 0.95;
var stepWidth = innerWidth / (rowCount - (1 - bandPercent));
var bandWidth = bandPercent * stepWidth;

var offsetCenterYPercent = 100;

// X coordinates of title/detail are relative to the center of the chart, as per option `offsetCenter`.
// Also, center points on each band, by offsetting by half band.
var offsetCenterX0 = -innerWidth / 2 + bandWidth / 2;

for(var i = 0; i < rowCount; i++) {
  var offsetCenterX = offsetCenterX0 + stepWidth * i;

  records.push({
    // ...
    
    title: {
      offsetCenter: [offsetCenterX, offsetCenterYPercent + "%"],
      overflow: "truncate",
      width: titleWidth
    },
    detail: {
      offsetCenter: [offsetCenterX, (offsetCenterYPercent + 15) + "%"],
      width: titleWidth
    }
  });
}
```